### DISCUSSION TITLE
`tee` will convert input to list, is that by design? (fixed in nushell 0.98.0)

### DISCUSSION THREAD

**Discussion by wdv4758h** [nushell/nushell#12715] [Q&A]
![image](https://github.com/nushell/nushell/assets/2716047/605ca3e9-9d6a-4173-906f-63ac3677733d)

**Comment by fdncred** [comment]
Seems normal to me. One doesn't typically `tee` an int, but maybe I'm wrong.

/cc the author @devyn for clarity.

**Comment by wdv4758h** [comment]
Not only int. A record will become table after tee. A string will become list after tee.

It feels like a suprise to me. As tee means copy data, but the copy change the data.

My use case is branching out string or record from the pipeline and save the value as cache. It serves as half processing cache, while original data will keep moving to following steps. But since tee will mess with the type, I will need to choose either "hack it. get the first element from list after tee", or "just don't use tee".

I use something like `do -c { let data = $in; ...; $data }` before there is `tee` in nushell.

**Reply by devyn** [reply]
It is possible to fix this, but one issue is that we have nowhere to inject an error into. For the stream, we inject it back into the original stream. I think I'll probably have to just settle for logging any error produced to stderr.

I do agree that it probably should pass a plain Value through without streaming it. It's a totally different implementation to do so, but it makes sense.

**Comment by suimong** [comment]
Stumbled on this today as well as the type changing behavior of `tee` caught me by surprise. Use case is similar to @wdv4758h  in that the value piped out from `tee` will be further processed down the pipeline therefore the type-changing behavior breaks user expectation.

**Reply by wdv4758h** [reply]
The type changing is really annoying to me. So I created my own `mytee` and use it everywhere now.

My code that put in the tee is not that complicated. Mostly, I just make sure it does not have error inside. And print it out or run it outside if I need more debug info. But it usually not that bad, it still shows errors if happened.

Not a elegant way. But at least it won't break my code.

**Comment by wdv4758h** [comment]
For anyone who accidentally view this thread in the future. The `tee` seems to be fixed in nushell 0.98.0. Ref: https://github.com/nushell/nushell/pull/13652
